src/index.ts

```diff
@@ -139,6 +139,24 @@ export function isArrayOf<T> (itemGuard: Guard<T>): Guard<T[]> {
   return ((input: any) => Array.isArray(input) && input.every(itemGuard)) as Guard<T[]>
 }
 
+const optionalSymbol = Symbol('optional')
```

Comment:
I'm not a big fan of using a symbol for identify "optional type guards" as it means only type guards that go through the "optional" function defined by this library are considered "optional type guards".

This symbol could possibly be exported so that others can use it but I think it would be best to remove it all together. However to remove it, an alternative implementation of `isOptional` or the part of `isOfShape` that uses `isOptional` would be needed.

Reply:
One possible solution would be to introduce a new type for `Guard`s with options and allow this type to be used in a `Shape`. "optional" could then simply be an option for guards.

The type could possibly just be a tuple like this:

```
type GuardWithOptions<T> = [Guard<T>, GuardOptions]
type GuardOptions = {
  optional?: boolean
}
```

Reply:
I see the mentioned drawbacks with the symbol, but I think it's fine for this library. Something more robust (like a full-blown validation library) would probably benefit from a more complex approach; `type-guards` is supposed to be a quick utility to help with basic validation (“binary” validation I guess, since it's either a pass or fail, i.e. no specific errors/messages of _why_ it failed).